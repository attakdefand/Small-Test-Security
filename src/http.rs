use axum::{routing::get, Router};
use serde::Serialize;

#[derive(Serialize)]
struct Health { status: &'static str, version: &'static str }

pub fn app() -> Router {
    Router::new().route("/health", get(|| async {
        axum::Json(Health { status: "ok", version: "v1" })
    }))
}
