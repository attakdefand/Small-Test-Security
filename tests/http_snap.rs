use axum::body::{to_bytes, Body};   // <-- import to_bytes here
use axum::http::Request;
use tower::util::ServiceExt;
use super_test_types_rust::http::app;

#[test]
fn health_snapshot() {
    let rt = tokio::runtime::Runtime::new().unwrap();
    rt.block_on(async {
        let res = app()
            .oneshot(Request::builder().uri("/health").body(Body::empty()).unwrap())
            .await
            .unwrap();

        assert_eq!(res.status(), http::StatusCode::OK);

        // axum 0.7: to_bytes(body, limit)
        let bytes = to_bytes(res.into_body(), usize::MAX).await.unwrap();
        let json = String::from_utf8(bytes.to_vec()).unwrap();

        insta::assert_json_snapshot!("health_v1", json);
    });
}
