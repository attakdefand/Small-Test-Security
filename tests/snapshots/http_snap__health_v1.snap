---
source: tests/http_snap.rs
expression: json
---
"{\"status\":\"ok\",\"version\":\"v1\"}"
